amida-tech/orange-ignite
# Feature/orange 90 go log metadata updating

If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### What does this PR do?
This allows the user to edit the fields listed under "Information" on the Log Details Page. This is accomplished by replacing the list with a FormList based on the LogDetailViewScreen state.

Most interesting edits are in LogDetailViewScreen.js.

#### Related JIRA tickets:
https://jira.amida-tech.com/browse/ORANGE-90

#### How should this be manually tested?

1. Log in as a user with access to a log.
2. Go to the log details screen for that log.
---
1. Press "Edit"
2. Verify list changes and all fields except "Log Owner" are editable
3. Make a change and press "Done"
4. Verify prompt pops up asking user to save or discard changes
5. Press "Discard"
6. Verify changes were not saved
---
1. Press "Edit"
2. Don't change anything
3. Press "Done"
4. Verify no prompt is shown before switching back to not editable list
---
1. Press "Edit"
2. Change something
3. Press "Done"
4. Verify prompt pops up asking user to save or discard changes
5. Press "Save"
6. Verify in reactotron that a successful response from the api was received
7. Verify the edits you made are kept in the list
8. Go back to Logs screen
9. Press on log to go to log details screen again
10. Verify the edits you made are still present

#### Background/Context
For now this just uses text boxes without client side validation. A future form update will prompt an update of this feature to be more user friendly.

#### Screenshots (if appropriate):
Not Editable
<img src="https://user-images.githubusercontent.com/5448265/36918018-79bef292-1e26-11e8-8b08-7b603c7a5dfd.png" width=300>

Editable
<img src="https://user-images.githubusercontent.com/5448265/36918024-805ac540-1e26-11e8-80a2-9716e27da751.png" width=300>

Prompt
<img src="https://user-images.githubusercontent.com/5448265/36918029-83d23a32-1e26-11e8-843c-7b7860482eba.png" width=300>
